docs/overview.rst

```diff
@@ -193,30 +195,22 @@ Addresses
 
 .. py:method:: Web3.isAddress(value)
 
-    Returns ``True`` if the value is one of the recognized address formats.
+    Returns ``True`` if the value is a valid `EIP55`_ checksummed address.
+    Note that EIP55 requires the '0x' prefix.
```

Comment:
My initial reaction is that the functionality of `Web3.isAddress` should not change.

It's still a useful thing to be able to check if something is an address in the basic definition of address as a hex encoded 20 byte value.  With this change users will have to resort to digging into `eth_utils` if they want this functionality and I don't think that's a good thing.